### PR TITLE
Material component documentation

### PR DESCRIPTION
**Pull request by gadams3**
Signed-off-by: Guthrie Adams <guthadam@amazon.com>

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/components/reference/atom/material.md [hunk 1]
```diff
@@ -1,20 +1,102 @@
...
 title: Material Component
 linktitle: Material
-description: 'Open 3D Engine (O3DE) Material component reference.'
+description: 'Open 3D Engine (O3DE) Material Component reference.'
 toc: true
 ---
 
-The **Material** component adds a material on the object's mesh. The material can be edited using the Material Editor.
-
+The **Material** component is used to inspect, customize, and otherwise override materials and their properties on entities with compatible renderable components. [Mesh Component](/docs/user-guide/components/reference/atom/mesh/) and [Actor Component](/docs/user-guide/components/reference/atom/actor/) are currently the only components affected by the Material Component. However, any component can support the Material Component by implementing the required buses and services. The material assets used by the Material Component are most commonly created using the Material Editor. They are also produced by the asset processor and model asset builder, which exports the materials defined within a source model, like an FBX file.
```
Comment (galibzon) [inline content/docs/user-guide/components/reference/atom/material.md:8]: The Actor component link is broken.

**Review by galibzon — APPROVED** [review]

**Review by chanmosq — CHANGES_REQUESTED** [review]
Started review - clarified some sentences and left a few questions

File: content/docs/user-guide/components/reference/atom/material.md [hunk 2]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
...
+| The context menus and large button at the top of the Material Component give access to the 'Generate/Manage Source Materials' dialog. The dialog displays a list of the unique materials associated with each material slot by default. After configuring target filenames, the dialog will be able to generate and automatically assign source material files to every matching material slot. The source materials can then be opened and edited with the Material Editor as well as shared with other entities. This window predates the existence of the Material Instance Inspector, the ability to customize materials without assigning a material asset, and the more robust model exporter that exists today. Using the dialog is not required but can simplify automatically assigning materials shared by multiple entities. It can also be useful for building a material library based on materials embedded in a FBX or other source. If source materials are not required for your workflow then materials can be customized and saved directly on entities using the Material Instance Inspector. | ![material-componen-generate](/images/user-guide/components/reference/atom/material/material-generate-ui.png) |
```
[the remark this answers is not in the document]
Comment (gadams3) [reply content/docs/user-guide/components/reference/atom/material.md:30]: It initializes with default filenames based on the material slot name. Users can optionally select their own file names. We may replace this workflow with one that just selects a folder for all of the materials.

Which workflow? If the materials were previously exported from this dialog and customized in the material editor then they can be applied to any other entity. If multiple entities are selected when this dialog is opened from the component context menu, and they all share the same set of material slots, the generated materials will be assigned on every entity.

File: content/docs/user-guide/components/reference/atom/material.md [hunk 3]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
 toc: true
 ---
 
-The **Material** component adds a material on the object's mesh. The material can be edited using the Material Editor.
-
+The **Material** component is used to inspect, customize, and otherwise override materials and their properties on entities with compatible renderable components. [Mesh Component](/docs/user-guide/components/reference/atom/mesh/) and [Actor Component](/docs/user-guide/components/reference/animation/actor/) are currently the only components affected by the Material Component. However, any component can support the Material Component by implementing the required buses and services. The material assets used by the Material Component are most commonly created using the Material Editor. They are also produced by the asset processor and model asset builder, which exports the materials defined within a source model, like an FBX file.
```
[the remark this answers is not in the document]
Comment (chanmosq) [reply content/docs/user-guide/components/reference/atom/material.md]: Ah yes that's key, my latest suggestion will address that.

File: content/docs/user-guide/components/reference/atom/material.md [hunk 4]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
 toc: true
 ---
 
-The **Material** component adds a material on the object's mesh. The material can be edited using the Material Editor.
-
+Use the **Material** component to inspect, customize, and override materials and properties used by other compatible components. Currently, only the [Mesh component](/docs/user-guide/components/reference/atom/mesh/) and the [Actor component](/docs/user-guide/components/reference/animation/actor/) support the Material component. However, any component can support the Material component by implementing the required buses and services. The Material component uses material assets, which you can create and configure using the Material Editor. Another way that material assets are produced is by the Asset Processor and the Model Asset Builder. This exports the materials defined within a source model, like an FBX file.
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/material.md:8]:
```suggestion
Use the **Material** component to inspect, customize, and override materials and properties on entities with compatible render components. Currently, only the [Mesh component](/docs/user-guide/components/reference/atom/mesh/) and the [Actor component](/docs/user-guide/components/reference/animation/actor/) support the Material component. However, any component can support the Material component by implementing the required buses and services. The Material component uses material assets, which you can create and configure using the Material Editor. Another way that material assets are produced is by the Asset Processor and the Model Asset Builder. This exports the materials defined within a source model, like an FBX file.
```

File: content/docs/user-guide/components/reference/atom/material.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chanmosq) [reply content/docs/user-guide/components/reference/atom/material.md:30]: That answers it perfectly, thanks!

**Review by chanmosq — CHANGES_REQUESTED** [review]
Mentioned this in another comment, but I think an important concept we need to grasp when using the many material editing tools is what the difference is between "source material" and "material instances". There's some small hints to it, but it's not concretely defined or introduced. Getting crisp and being consistent with the terminology is important to lay the foundational understanding of materials in O3DE. 

(Note: "Material Instance" is defined in the [Atom Glossary](https://www.o3de.org/docs/atom-guide/glossary/), if you want to pull from that.)

File: content/docs/user-guide/components/reference/atom/material.md [hunk 5]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
...
+| The Material Instance Inspector provides all of the same features as the inspector found in the Material Editor. Instead of creating a child material, you have the option of customizing material properties that override values from the asset assigned to a material slot. The changes will be applied to the entity in real time, visible in the viewport and previews. This window will remain opened and locked to the entity and material slot selection until manually closed. The UI will be disabled if the entity or the selected material slot become invalid or unavailable. | ![material-componen-inspector](/images/user-guide/components/reference/atom/material/material-inspector-ui.png) |
+
+| Property | Description | Values | Default |
+|-|-|-|-|
+| **Generate/Manage Source Materials...** | This button opens the "Generate/Manage Source Materials" dialog. | | |
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/material.md:35]:
```suggestion
| **Generate/Manage Source Materials...** | This button opens the **Generate/Manage Source Materials** dialog. | | |
```

File: content/docs/user-guide/components/reference/atom/material.md [hunk 6]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
...
+| **Default Material** | Materials assigned to the default material slot will be applied to the entire model wherever no higher priority model or LOD material is assigned. | Material Asset | None |
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/material.md:36]:
```suggestion
| **Default Material** | The default material applies to the entire model, except for parts where a higher priority model or LOD material is assigned. | Material Asset | None |
```

File: content/docs/user-guide/components/reference/atom/material.md [hunk 7]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
...
+| **Model Materials** | Materials assigned to these slots will be applied to every part of the model with the same material slot name unless a higher priority LOD material is assigned. | Container of material assets | Empty |
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/material.md:37]:
```suggestion
| **Model Materials** | Materials that are assigned to these slots are applied to every part of the model with the same material slot name, unless a higher priority LOD material is assigned. | Container of material assets | Empty |
```

File: content/docs/user-guide/components/reference/atom/material.md [hunk 8]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
...
+| **Enable LOD Materials** | When this flag is enabled, materials can be customized per LOD. | Boolean | `Disabled` |
+| **LOD Materials** | Materials assigned to LOD material slots will override model and default material assignments for the given LOD. If you are changing the default material or model materials and not seeing those changes applied then verify that there is no LOD material taking higher priority or disable them completely. | Container of material assets organized by LOD | Empty |
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/material.md:39]:
```suggestion
| **LOD Materials** | Materials assigned to LOD material slots override **Default Material** and **Model Material** assignments for the given LOD. If you change the default material or model materials and don't see those changes applied, then verify that there's no LOD material taking higher priority or disable them completely. | Container of material assets organized by LOD | Empty |
```

File: content/docs/user-guide/components/reference/atom/material.md [hunk 9]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
...
+
+## MaterialComponentRequestBus
+This request bus supplies the interface for interacting with the Material Component. Many of the functions relate to inspecting or customizing material asset and property overrides. All of the functions are available in the editor, gameplay, and simulation. Many of them may be more useful for extending Material Component related tools and scripts.
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/material.md:42]:
```suggestion
This request bus supplies the interface for interacting with the Material component. It contains many of the functions that relate to inspecting or customizing material asset and property overrides. All of the functions are available in the editor, gameplay, and simulation. Many of them may be more useful for extending Material component related tools and scripts. 
```

**Review by chanmosq — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/components/reference/atom/material.md [hunk 10]
```diff
@@ -5,16 +5,98 @@ description: 'Open 3D Engine (O3DE) Material component reference.'
...
+| **Enable LOD Materials** | When this flag is enabled, materials can be customized per LOD. | Boolean | `Disabled` |
+| **LOD Materials** | Materials assigned to LOD material slots will override model and default material assignments for the given LOD. If you are changing the default material or model materials and not seeing those changes applied then verify that there is no LOD material taking higher priority or disable them completely. | Container of material assets organized by LOD | Empty |
+
+## MaterialComponentRequestBus
+This request bus supplies the interface for interacting with the Material Component. Many of the functions relate to inspecting or customizing material asset and property overrides. All of the functions are available in the editor, gameplay, and simulation. Many of them may be more useful for extending Material Component related tools and scripts. 
+
+| Method Name | Description | Parameter | Return | Scriptable |
+|-|-|-|-|-|
+| 'GetOriginalMaterialAssignments' | Returns a map of available material slots and their default assignments | None | Unmodified Material Slot Layout: MaterialAssignmentMap | Yes |
```
Comment (chanmosq) [inline content/docs/user-guide/components/reference/atom/material.md:46]:
```suggestion
| 'GetOriginalMaterialAssignments' | Returns a map of available material slots and their default assignments. | None | Unmodified Material Slot Layout: MaterialAssignmentMap | Yes |
```

Missing a period here and other method descriptions.

**Comment by gadams3** [comment]
> Mentioned this in another comment, but I think an important concept we need to grasp when using the many material editing tools is what the difference is between "source material" and "material instances". There's some small hints to it, but it's not concretely defined or introduced. Getting crisp and being consistent with the terminology is important to lay the foundational understanding of materials in O3DE.
> 
> (Note: "Material Instance" is defined in the [Atom Glossary](https://www.o3de.org/docs/atom-guide/glossary/), if you want to pull from that.)

is there a way to just link to that?

**Comment by chanmosq** [comment]
> > Mentioned this in another comment, but I think an important concept we need to grasp when using the many material editing tools is what the difference is between "source material" and "material instances". There's some small hints to it, but it's not concretely defined or introduced. Getting crisp and being consistent with the terminology is important to lay the foundational understanding of materials in O3DE.
> > (Note: "Material Instance" is defined in the [Atom Glossary](https://www.o3de.org/docs/atom-guide/glossary/), if you want to pull from that.)
> 
> is there a way to just link to that?

The glossary definition of Material Instance is only part of the equation. I think the concept that's missing is the fact there is a distinction between the two, and their definition. To understand the material features, we also need to understand the terminology used. Would you be able to provide a definition for source material and add it to the glossary?

**Comment by gadams3** [comment]
> > > Mentioned this in another comment, but I think an important concept we need to grasp when using the many material editing tools is what the difference is between "source material" and "material instances". There's some small hints to it, but it's not concretely defined or introduced. Getting crisp and being consistent with the terminology is important to lay the foundational understanding of materials in O3DE.
> > > (Note: "Material Instance" is defined in the [Atom Glossary](https://www.o3de.org/docs/atom-guide/glossary/), if you want to pull from that.)
> > 
> > 
> > is there a way to just link to that?
> 
> The glossary definition of Material Instance is only part of the equation. I think the concept that's missing is the fact there is a distinction between the two, and their definition. To understand the material features, we also need to understand the terminology used. Would you be able to provide a definition for source material and add it to the glossary?

Source material is just the .material JSON file that can be opened in the material editor. We just started calling it that as an abbreviation for “.material source file”. That's also to distinguish them from “.azmaterial product asset”, the files output from the asset processor that show up in the asset browser. All assets have source files, the files fed into the asset processor, and product assets, the files output from the asset processor. Source and product assets should be documented in the asset system. 

The generate/manage materials dialog might need to be renamed. It does the opposite job of the asset processor. It converts product .azmaterial assets, that were produced using a different source file type, like FBX files, into .material source files that might not have existed.

Material instances are the active materials applied to models, as described. In the context of the material component, there is one material instance for each material slot. the material instance will be created based on which asset is being used. If the material component slot has custom properties for that instance then it must create a unique instance to apply those properties. if there are no unique properties set, it will share a material instance with anything else using the same material.

**Review by chanmosq — APPROVED** [review]

**Pull request merged**